core/domain/acl_decorators.py

```diff
@@ -77,7 +118,8 @@ def test_can_play(self, exploration_id, **kwargs):
                 exploration_id):
             return handler(self, exploration_id, **kwargs)
         else:
-            raise self.PageNotFoundException
+            self.redirect('/')
+            return
```

Comment:
Check for some particular text on the page? PageNotFound errors lead to 404s which have specific, custom error pages.